tasks/Functions-client/deployAutoClient.js

```diff
@@ -36,42 +37,48 @@ task("functions-deploy-auto-client", "Deploys the AutomatedFunctionsConsumer con
       throw Error("Gas limit must be less than or equal to 300,000")
```

Comment:
Remove this gas limit check.  Instead, use `subManager.estimateFunctionsRequestCost`. This is because it is possible for certain subscriptions which we manually allow to use more than 300k gas.  This logic is all verified by estimateCost.  See here: https://github.com/smartcontractkit/functions-hardhat-starter-kit/blob/develop/tasks/Functions-client/request.js#L105